libsolidity/formal/CHC.cpp

```diff
@@ -196,8 +196,12 @@ void CHC::endVisit(ContractDefinition const& _contract)
 	connectBlocks(m_currentBlock, summary(_contract));
 
 	setCurrentBlock(*m_constructorSummaries.at(&_contract));
-	m_queryPlaceholders[&_contract].push_back({smtutil::Expression(true), errorFlag().currentValue(), m_currentBlock});
-	connectBlocks(m_currentBlock, interface(), errorFlag().currentValue() == 0);
+	auto constructor = _contract.constructor();
+	auto txConstraints = state().txConstraints();
```

Comment:
Where are these constraints added?

Reply:
Just below (lines 203 and 204), as part of the condition for the placeholder and for the edge connecting blocks.

Reply:
Ah I see, sorry I misunderstood. So `state().txConstraints()` existed but was only added in this PR. I was looking for the origin of `state().txConstraints()` :)

Reply:
Yes, the code existed before, and the constraints have already been added for regular transactions before, but they were missing from the deployment phase, to capture the fact that they must hold also when the transaction data are accessed in constructor.